### DISCUSSION TITLE
1.4.1

### DISCUSSION THREAD

**Discussion by kirb** [LegacyUpdate/LegacyUpdate#36] [Announcements]
* Works around a bug with the Windows Update Agent 7.6.7600.256 installer on Windows XP Home Edition, resulting in it failing with error code 82, and potentially rendering the system unable to boot (“NTLDR is missing” error). Version 7.6.7600.244 is now installed instead on Home Edition.
* Fixes the setup not automatically continuing after a restart, and not deleting the LegacyUpdateTemp user after it’s completed.

Huge thanks to those who reported and helped me to find the root cause of the Home Edition issue.

<hr /><em>This discussion was created from the release <a href='https://github.com/kirb/LegacyUpdate/releases/tag/v1.4.1.0'>1.4.1</a>.</em>

**Comment by S1m0n32002** [comment]
For some reason it's giving me an error window saying "Windows Update Agent Failed to download".

The system is running Windows XP Professional SP3.

![20221216_212302](https://user-images.githubusercontent.com/39035729/208183197-085ba339-c9ad-4662-988c-283f21714602.jpg)

On another system it just crashes but it might be a lack of resources due to the low specs and missing drivers.

**Reply by S1m0n32002** [reply]
UPDATE: I seem to have understood and fixed the cause of the problem.
The computer is the infamous Acer Aspire A0A110 with the slowest ssd in history.
Apparently if you have a very fast connection and a very slow storage medium it gets stuck somewhere and the download gets interrupted.

I solved the problem by connecting the computer via 115200 bauds serial and sharing my connection through it.

Then started throwing new errors.

Basically if you have to use the most painfully slow SSD ever existed random errors will appear.

I hope this helps anyone else in my situation :)

(And no, swapping that ssd is too much of a hassle. I've already gave up twice while trying)

**Reply by kirb** [reply]
Very interesting, I’d bet the SSD writes are so slow that the server is thinking the connection might have dropped, or possibly WinInet’s buffer fills up quicker than it’s able to write out the contents of the buffer to disk. No idea if there’s something that can be done about it, or if that’s just expected behavior of WinInet.

**Reply by S1m0n32002** [reply]
idk, but I wouldn't worry too much. Maybe just add a reminder or something just in case but it's really not that big of a deal.
I should have thrown away that SSD years ago. (Even Linus from LTT threw it in the bin.)

Anyway, it's been updating for like 3 days now and even though it's way more powerful that my other WinXP laptops (Pentium 3 m and a Celeron 800mhz both with 120mb of ram).

**Comment by DCFUKSURMOM** [comment]
Does not work on XP SP3, wont even launch the installer. Tested in a fresh install in QEMU.
![error](https://user-images.githubusercontent.com/87423173/208247031-eea08a68-93e1-4908-8279-d7f3e3de2c9f.png)
[74ca_appcompat.txt](https://github.com/kirb/LegacyUpdate/files/10251929/74ca_appcompat.txt)
